code/script.js

```diff
@@ -218,108 +227,180 @@ const generateBoard = () => {
+      );
+      alert(
+        `Yes, the person has ${value} hair! Keep all people that have ${value} hair.`
+      );
+    } else {
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] !== value
+      );
+      alert(
+        `No, the person doesn't have ${value} hair! Remove all people that have ${value} hair.`
+      );
+    }
+  
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] === value
+      );
+      alert(
+        `Yes, the person has ${value} eyes! Keep all people that have ${value} eyes.`
+      );
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] !== value
+      );
+      alert(
+        `No, the person doesn't have ${value} eyes! Remove all people that have ${value} eyes.`
+      );
     }
   }
+  generateBoard(keep);
+};
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
 
-  // Invoke a function to redraw the board with the remaining people.
-}
-
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+const guess = (personToCheck) => {
+  
+  let playerGuess = confirm(`Do you want to guess on this fellow ${personToCheck}?`);
+  
+  if (playerGuess) {
+    checkMyGuess(personToCheck);
+  } else {
+    alert("You can continue with the game!");
+  }
+};
 
-// If you confirm, this function is invoked
+// If the player confirms, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerHTML = `You won! You guessed on the correct person - ${personToCheck}!`;
+  } else {
+    winOrLoseText.innerHTML = ` Bad guess! this was not the right person. It was ${secret.name}.`;
```

Comment:
also here that it displays the person who was the secret one when you guessed wrong. Nice!